### PR TITLE
Calculate events

### PR DESCRIPTION
**Pull request by SamTush**
## What does this PR do?

This PR is raised to implement the following features: 👍 :

- [x] Handle events in a React app.
- [x] The application runs in the browser without errors 
- [x] No linter errors
- [x] Use Gitflow

## Description of the task to be completed ↓↓ 

- Handle events in the calculator.

  
## How should it be tested

Clone the repository
```
git clone https://github.com/SamTush/math-magicians.git
```
Navigate to the folder
```
cd math-magicians
```
Checkout the branch
```
git checkout feature/calculator-events
```

## How should it be run

Run application in browser
```
run start
```

### CODE REVIEW

File: src/components/calculator.js [hunk 1]
```diff
@@ -1,99 +1,88 @@
...
+  const { total, next, operation } = currentOperation;
   return (
     <div className="main-cal p-4 container-fluid d-flex justify-content-center">
       <div className="row">
         <div className="col ps-3 pe-3 pt-5 pb-4 col-12 calculator">
           <table className="fit">
             <tr>
               <td colSpan={4}>
-                <input type="text" className="output mb-1 p-2" id="output" value="0" />
+                <p className="output mb-1 p-2">{ next || operation || total || 0}</p>
```
Comment (nahnahdev) [inline src/components/calculator.js:25]: - [ ] Great job so far @SamTush your project looks great :+1: However, I noticed a minor bug. The `0` button doesn't work. So if I want to divide 1000 by 8, there is no way of doing that. So kindly fix this so your app can work perfectly and meet all the requirements :+1: 
- Video Reference
[Screencast from 03-30-2023 02:45:37 PM.webm](https://user-images.githubusercontent.com/102699958/228874372-48ec0aca-d5e5-4687-a7d2-8a6552a4affd.webm)

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @SamTush , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- The Calculator app handles state with `useState` :heavy_check_mark: 
- Click events are implemented for almost all buttons :heavy_check_mark:
- The project has an `operate.js` file copied from [this one](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js). :heavy_check_mark: 
- The project has a `calculate.js` file copied from [this one](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js). :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SamTush** [comment]
Thank you @NahnahAJ  for reviewing my PR👍🏽. Happy coding😎

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @SamTush , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- App works as expected :heavy_check_mark: 
- Bug fixed :100: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by SamTush** [comment]
Thank you for reviewing my PR review @NahnahAJ 😎. Cheers and happy coding

**Pull request merged**